## Sharpe Ratio Analysis -> RELIANCE & IOC (Dec 2025 – Apr 2026)

In [1]:
import yfinance as yf
import pandas as pd

risk_free = 0.0625 / 252
sharpe = {}

stocks = ['RELIANCE.NS', 'IOC.NS']
df = yf.download(stocks, start='2025-12-01', end='2026-04-24', progress=False)['Close']
stock_returns = df.pct_change().dropna()

for stock in stock_returns.columns:
    excess_return = stock_returns[stock].mean() - risk_free
    sharpe_ratio = excess_return / stock_returns[stock].std()
    sharpe[stock] = round(sharpe_ratio, 2)

print(sharpe)

{'IOC.NS': -0.04, 'RELIANCE.NS': -0.12}


Honestly this result caught me off guard a little.

Both stocks came out with a negative Sharpe Ratio which basically 
means during this entire geopolitical crisis period, neither RELIANCE 
nor IOC gave returns good enough to even beat a simple FD rate of 6.25%. 
Like you would have literally made more money just putting cash in a 
bank than holding these stocks.

**IOC at -0.04** is barely negative so it was close but just not 
enough. The government controls on fuel prices probably saved it 
from being worse.

**RELIANCE at -0.12** was the bigger surprise for me. I expected 
Reliance to hold up better because of Jio and retail but clearly 
the oil side of the business dragged everything down during this period.

But here is the thing — this does not mean these are bad stocks. 
It just means this was a bad time to hold them. The geopolitical 
uncertainty created so much volatility that the risk taken was just 
not worth the returns during these 4 months specifically.

This is actually the whole point of calculating Sharpe Ratio in algo 
trading — it tells you not just how much you made but whether the 
risk you took was actually worth it. And in this case, it wasn't.